src/pkg/repository/repository_test.go

```diff
@@ -284,13 +284,88 @@ func (suite *RepositoryIntegrationSuite) TestConnect_DisableMetrics() {
 	// need to initialize the repository before we can test connecting to it.
 	st := tester.NewPrefixedS3Storage(t)
 
-	repo, err := repository.Initialize(ctx, account.Account{}, st, control.Defaults())
+	repo, err := Initialize(ctx, account.Account{}, st, control.Defaults())
 	require.NoError(t, err)
 
 	// now re-connect
-	r, err := repository.Connect(ctx, account.Account{}, st, repo.GetID(), control.Options{DisableMetrics: true})
+	r, err := Connect(ctx, account.Account{}, st, repo.GetID(), control.Options{DisableMetrics: true})
 	assert.NoError(t, err)
 
 	// now we have repoID beforehand
 	assert.Equal(t, r.GetID(), r.GetID())
 }
+
+// Test_Options tests that the options are passed through to the repository
+// correctly
+func (suite *RepositoryIntegrationSuite) Test_Options() {
```

Comment:
I'm undecided on whether this is the correct location and owner for this test.  Do we expect repo connection to modify options or extensions?  Do we expect repo connection to interact with those details at all?

This feels like a basic test for existence within a consumer.  "I added foo to bar.  Does bar contain foo?  Yes."  It's testing, yes, but I don't see the assurance guarantees we're getting out of it.  Perhaps there's a lower-level test, more associated with usage and interaction of the options, that would make more sense?

Reply:
>  "I added foo to bar. Does bar contain foo? Yes."

Yep. This is the intent of this particular test. I'll be adding more tests which exercise options usage

Reply:
> This is the intent of this particular test

Right, so you're testing struct population.  I don't see that as providing value; it's too foundational of a behavior to need to test.  Unless, of course, the consumer you're handing the struct into is mutating it in some way.

Reply:
This is just to protect against future regressions in `Initialize` and `Connect`